web/themes/custom/usagov/usagov.theme

```diff
@@ -10,13 +10,248 @@ use Drupal\taxonomy\Entity\Term;
  * Place your custom PHP code in this file.
  */
 
-/**
-* This makes the node varialbe available to the templates/html.html.twig template file
-*/
 function usagov_preprocess_html(&$variables) {
-  $variables['node'] = \Drupal::routeMatch()->getParameter('node');
+  $node = \Drupal::routeMatch()->getParameter('node');
+  $variables['node'] = $node;
+
+  // If the FAQ checkbox is checked, then parse the content to find questions and answers
+  if ($node?->hasField('field_faq_page') && $node->get('field_faq_page')[0]?->value) {
+    $variables['faq'] = generate_faq($node);
+  }
+}
+
+/**
+ * generate_faq takes a node, analyses the body content,
+ * and returns an array of objects each containing a question and answer
+ */
+function generate_faq(&$node) {
+  $faq = [];
+
+  // Set up a DOMDocument with the node's body content
+  $node_body = $node->get('body')[0]->value;
+  $dom = new DOMDocument();
+  libxml_use_internal_errors(TRUE);
+  $dom->loadHTML($node_body);
```

Comment:
I caught a character encoding issue on the /real-id example, and I have the solution: Every time you call `$dom->loadHTML` on some content from the document, prepend `'<?xml encoding="UTF-8">' . ` to that content. For example: 
```
$dom->loadHTML('<?xml encoding="UTF-8">' . $node_body);
```

We need this everywhere you call loadHTML here except for the empty elements (like the one on line 44). I found this in the top comment here: https://www.php.net/manual/en/domdocument.loadhtml.php and I used it when importing the federal directory records. 

The thing that caught my eye in the real-id example is an apostrophy in the first sentence on that page.

Reply:
I added it to the empty ones too assuming that it wont hurt anything.